src/python/pants/backend/python/rules/pytest_runner.py

```diff
@@ -228,26 +240,47 @@ async def run_python_test(
     test_options: TestOptions,
 ) -> TestResult:
     """Runs pytest for one target."""
-    env = {"PYTEST_ADDOPTS": f"--color={'yes' if global_options.options.colors else 'no'}"}
+    add_opts = [f"--color={'yes' if global_options.options.colors else 'no'}"]
+    test_results_file = f"{config.address.path_safe_spec}.xml"
+    if test_setup.xml_dir:
+        add_opts.extend(
+            (f"--junitxml={test_results_file}", f"-o junit_family={test_setup.junit_family}",)
+        )
+    env = {"PYTEST_ADDOPTS": " ".join(add_opts)}
+
     run_coverage = test_options.values.run_coverage
+    output_dirs = [".coverage"] if run_coverage else []
+    if test_setup.xml_dir:
+        output_dirs.append(test_results_file)
     process = test_setup.test_runner_pex.create_process(
         python_setup=python_setup,
         subprocess_encoding_environment=subprocess_encoding_environment,
         pex_path=f"./{test_setup.test_runner_pex.output_filename}",
         pex_args=test_setup.args,
         input_files=test_setup.input_files_digest,
-        output_directories=(".coverage",) if run_coverage else None,
+        output_directories=tuple(output_dirs) if output_dirs else None,
         description=f"Run Pytest for {config.address.reference()}",
         timeout_seconds=(
             test_setup.timeout_seconds if test_setup.timeout_seconds is not None else 9999
         ),
         env=env,
     )
     result = await Get[FallibleProcessResult](Process, process)
-    coverage_data = (
-        PytestCoverageData(config.address, result.output_directory_digest) if run_coverage else None
+    output_digest = result.output_directory_digest
+
+    coverage_data = PytestCoverageData(config.address, output_digest) if run_coverage else None
```

Comment:
How come you got rid of the snippet to remove the XML results from `coverage_data` via `SnapshotSubset`? Does that still work properly? 

Note that we do not have any tests for coverage yet. Please run `./v2 --test-run-coverage test src/python/pants/util:tests` to ensure that it still works properly.

Reply:
@Eric-Arellano I don't follow... what do u think I removed?
I just extracted `result.output_directory_digest` into a local variable (`output_digest`) which makes the expression shorter so black consolidated it into a single line...
<img width="935" alt="Screen Shot 2020-04-27 at 6 24 02 PM" src="https://user-images.githubusercontent.com/1268088/80436556-474de700-88b4-11ea-89a1-3cdde3d0c327.png">